aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
+
+
+array : '[' identifier_list ']'
+
+assignment_statement: identifier type '=' expression
+                    | '!' assignment_statement
+
+if_statement: 'if' expression NEWLINE body
+
+switch_statement: 'switch' identifier cases
```

Comment:
`cases` doesn't exist as a rule. Should just be a series of `case` rules (0 or more) and an optional `default` case (`default` being the keyword).